### DISCUSSION TITLE
Uma duvida sobre js `${const}${const}`

### DISCUSSION THREAD

**Discussion by HigorAln** [frontendbr/forum#2136] [Perguntas e Respostas] [answered]
Salve...
Comecei a estudar programação a pouco tempo e gostei muito da area do front, estou com uma duvida muito tensa e não acho resposta em canto nenhum, se alguem soube me responder, agradeço desde já...

Já vi umas 3-4 vezes isso `${const}${const}`(Esse 'const' e definindo o valor qualquer de uma const) porem não faço ideia oque e isso, e como funciona, então ja fico perdido no assunto quando isso aparece, alguem poderia por favor me tirar essa duvida.

**Comment by vitebo** [comment, marked as answer]
@HigorAln isso é `template strings` do JS, aqui eles explicam melhor o que é, pra que serve e como funciona:
- https://developer.mozilla.org/pt-BR/docs/Web/JavaScript/Reference/Template_literals
- https://developers.google.com/web/updates/2015/01/ES6-Template-Strings

**Reply by HigorAln** [reply]
Muito obg!!

**Comment by yurimutti** [comment]
Basicamente elas permitem expressões embutidas.

Exemplo:
const num1 = 1
const num2 = 2

Somando as variáveis e incluido a string total no final:

Template strings
const soma = `Soma é: ${num1 + num2} total`

Sem
const soma2 = 'Soma é: ' + num1 + num2 + ' total'

Percebe como facilita a escrita?

**Reply by HigorAln** [reply]
Entendi perfeitamente, muito muito obrigado!!
